dm/syncer/syncer.go

```diff
@@ -2092,14 +2173,18 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 
 		var originSQL string // show origin sql when error, only ddl now
 		var err2 error
+		var sourceTable *filter.Table
 
 		switch ev := e.Event.(type) {
 		case *replication.RotateEvent:
 			err2 = s.handleRotateEvent(ev, ec)
 		case *replication.RowsEvent:
 			eventIndex++
 			metrics.BinlogEventRowHistogram.WithLabelValues(s.cfg.WorkerName, s.cfg.Name, s.cfg.SourceID).Observe(float64(len(ev.Rows)))
-			err2 = s.handleRowsEvent(ev, ec)
+			sourceTable, err2 = s.handleRowsEvent(ev, ec)
+			if sourceTable != nil && err2 == nil && s.cfg.EnableGTID {
+				affectedSourceTables[sourceTable.String()] = struct{}{}
```

Comment:
hope it will no affect performance, or we may use sourceTable as the key instead of string

Reply:
IMI, in most times big transactions don't have many tables. If we save sourceTable pointer here it will cost us more in this kind of situation.
I have updated this to a two-dimensional map in https://github.com/pingcap/tiflow/pull/5198/commits/6e5bd635a07080d10dc9202175edcdb6bcd657d1, PTAL